src/adapter/graphql/src/queries/datasets/dataset.rs

```diff
@@ -98,6 +98,11 @@ impl Dataset {
         DatasetMetadata::new(self.dataset_handle.clone())
     }
 
+    /// Access to the environment variable of this dataset
+    async fn env_vars(&self) -> DatasetEnvVars {
+        DatasetEnvVars::new(self.dataset_handle.clone())
```

Comment:
Don't we need to check read permissions?
Also, to view secrets, I think we need a separate permission.

Reply:
read permission check is inside `dataset_env_vars.rs`